src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -579,28 +577,70 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
      * Finds and returns first sentence.
      *
      * @param ast Javadoc root node.
+     * @param period Period character.
      * @return first sentence.
      */
-    private static String getFirstSentence(DetailNode ast) {
+    private static String getFirstSentence(DetailNode ast, String period) {
         final StringBuilder result = new StringBuilder(256);
-        final String periodSuffix = DEFAULT_PERIOD + ' ';
-        for (DetailNode child : ast.getChildren()) {
-            final String text;
-            if (child.getChildren().length == 0) {
-                text = child.getText();
+        final boolean foundEnd = appendFirstSentence(ast, period, result);
```

Comment:
I do not see reason in this new method, fact that it takes result as parameter also looks weird, please inline it here.

Reply:
It's a recursive call to build the result. This line kicks off the recursion. I think the only way to inline it would be to replace the recursion with a loop. Do I need to do that?

Reply:
I see it now. Ok let me thinks on this.

Meanwhile please make all CIs green.

Reply:
thanks a lot.